langchain/src/vectorstores/closevector/common.ts

```diff
@@ -0,0 +1,226 @@
+      (x) => [
+        new Document({
+          pageContent: x[0].pageContent,
+          metadata: x[0].metadata || {},
+        }),
+        x[1],
+      ]
+    );
+    return mapped;
+  }
+
+  /**
+   * Method to delete the vector store from a directory. It deletes the
+   * hnswlib.index file, the docstore.json file, and the args.json file from
+   * the directory.
+   * @param params An object with a directory property that specifies the directory from which to delete the vector store.
+   * @returns A Promise that resolves when the vector store has been deleted.
+   */
+  async delete(params: { directory: string }) {
+    return await this.instance.delete(params);
+  }
+
+  static textsToDocuments(texts: string[], metadatas: object[] | object) {
+    const docs: Document[] = [];
+    for (let i = 0; i < texts.length; i += 1) {
+      const metadata = Array.isArray(metadatas) ? metadatas[i] : metadatas;
+      const newDoc = new Document({
+        pageContent: texts[i],
+        metadata,
+      });
+      docs.push(newDoc);
+    }
+    return docs;
+  }
+
+  /**
+   * Static method to create a new CloseVectorNode instance from texts and metadata.
+   * It creates a new Document instance for each text and metadata, then
+   * calls the fromDocuments method to create the HNSWLib instance.
+   * @param _texts The texts to be used to create the documents.
+   * @param _metadatas The metadata to be used to create the documents.
+   * @param _embeddings The embeddings to be used by the HNSWLib instance.
+   * @param _args An optional configuration object for the HNSWLib instance.
+   * @param _credential An optional credential object for the CloseVector API.
+   * @returns A Promise that resolves to a new CloseVectorNode instance.
+   */
+  static async fromTexts (
+    _texts: string[],
+    _metadatas: object[] | object,
+    _embeddings: Embeddings,
+    _args?: Record<string, unknown>,
+    _credential?: CloseVectorCredentials
+  ): Promise<CloseVector<CloseVectorSaveableVectorStore>> {
+    throw new Error("not implemented");
+  }
+
+  /**
+   * Static method to create a new CloseVectorNode instance from documents. It
+   * creates a new CloseVectorNode instance, adds the documents to it, then returns
+   * the instance.
+   * @param _docs The documents to be added to the HNSWLib instance.
+   * @param _embeddings The embeddings to be used by the HNSWLib instance.
+   * @param _args An optional configuration object for the HNSWLib instance.
+   * @param _credentials An optional credential object for the CloseVector API.
+   * @returns A Promise that resolves to a new CloseVectorNode instance.
+   */
+  static async fromDocuments(
+    _docs: Document[],
+    _embeddings: Embeddings,
+    _args?: Record<string, unknown>,
+    _credentials?: CloseVectorCredentials
+  ): Promise<CloseVector<CloseVectorSaveableVectorStore>> {
+    throw new Error("not implemented");
+  }
+
+  static async imports(): Promise<{
```

Comment:
What's the purpose of this method? Looks like it's not called?

Reply:
I was assuming LangChain might be used in some non-TypeScript environments that may not have solid type checking and tests. A class method in common.ts may be called directly. Thus, a runtime error is added to hint at such cases.

However, I forgot to check if there are any practices like this in LangChain. I have removed those methods in common.ts to adhere to the conventions.